File: java/src/edu/caltech/ipac/firefly/data/table/BaseTableColumn.java
package edu.caltech.ipac.firefly.data.table;


/**
 * Date: Nov 14, 2007
 *
 * @author loi
 * @version $Id: BaseTableColumn.java,v 1.13 2011/11/11 20:51:10 loi Exp $
 */
public class BaseTableColumn implements TableDataView.Column {
    private String title;
    private String name;
    private String type;
    private TableDataView.Align align;
    private int width;
    private int prefWidth = 0;
    private String units;
    private boolean isSortable = true;
    private boolean isHidden = false;
    private boolean isVisible = true;
    private String shortDesc;
    private String[] enumVals;
    private String[] sortByCols;
    private boolean requiresQuotes = false;

    public BaseTableColumn() {}

    public BaseTableColumn(String header) {
        this(header, TableDataView.Align.LEFT, 10, true);
    }

    public BaseTableColumn(String name, TableDataView.Align align, int width, boolean sortable) {
        this.name = name;
        this.title = name;
        this.shortDesc = name;
        this.align = align;
        this.width = width;
        this.isSortable = sortable;
    }

    public void setShortDesc(String shortDesc) {
        this.shortDesc = shortDesc;
    }

    public String getShortDesc() {
        return this.shortDesc;
    }

    public String getTitle() {
        return title;
    }

    public String getName() {
        return name;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setName(String name) {
        this.name = name;
    }

    public TableDataView.Align getAlign() {
        return align;
    }

    public void setAlign(TableDataView.Align align) {
        this.align = align;
    }

    /**
     * returns recommnended width for this columns.
     */
    public int getWidth() {
        return width;
    }

    /**
     * set the recommnended width for this columns.
     * @param width
     */
    public void setWidth(int width) {
        this.width = width;
    }

    public int getPrefWidth() {
        return prefWidth > 0 ? prefWidth : width;
    }

    public void setPrefWidth(int prefWidth) {
        this.prefWidth = prefWidth;
    }

    public String getUnits() {
        return units;
    }

    public void setUnits(String units) {
        this.units = units;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public boolean isSortable() {
        return isSortable;
    }

    public void setSortable(boolean sortable) {
        isSortable = sortable;
    }

    public boolean isHidden() {
        return isHidden;
    }

    public void setHidden(boolean hidden) {
        isHidden = hidden;
        isVisible = hidden ? false : isVisible;
    }

    public boolean isVisible() {
        return isVisible;
    }

    public void setVisible(boolean visible) {
        isVisible = visible;
    }

    public void setEnums(String[] enumVals) {
        this.enumVals = enumVals;
    }

    public String[] getEnums() {
        return enumVals;
    }

    public String[] getSortByCols() {
        return sortByCols;
    }

    public void setSortByCols(String[] secondarySortCols) {
        this.sortByCols = secondarySortCols;
    }

    public boolean isRequiresQuotes() {
        return requiresQuotes;
    }

    public void setRequiresQuotes(boolean requiresQuotes) {
        this.requiresQuotes = requiresQuotes;
    }


//====================================================================
//  private methods
//====================================================================

}

File: src/firefly/js/ipac-firefly/data/table/BaseTableColumn.js
/**
    * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
    * @author tatianag
    */

'use strict';

import Enum from "enum";

export const Align = new Enum(['LEFT', 'RIGHT', 'CENTER']);

export class BaseTableColumn {

    /*
     private String title;
     private String name;
     private String type;
     private TableDataView.Align align;
     private int width;
     private int prefWidth = 0;
     private String units;
     private boolean isSortable = true;
     private boolean isHidden = false;
     private boolean isVisible = true;
     private String shortDesc;
     private String[] enumVals;
     private String[] sortByCols;
     private boolean requiresQuotes = false;
     */

    constructor(name, align = Align.LEFT, width = 10, sortable = true) {
        this.name = name;
        this.title = name;
        this.shortDesc = name;
        this.align = align;
        this.width = width;
        this.isSortable = sortable;
    }

    setShortDesc(shortDesc) {
        this.shortDesc = shortDesc;
    }

    getShortDesc() {
        return this.shortDesc;
    }

    getTitle() {
        return this.title;
    }

    getName() {
        return this.name;
    }

    setTitle(title) {
        this.title = title;
    }

    setName(name) {
        this.name = name;
    }

    getAlign() {
        return this.align;
    }

    setAlign(align) {
        this.align = align;
    }

    /**
     * returns recommnended width for this columns.
     */
    getWidth() {
        return this.width;
    }

    /**
     * set the recommnended width for this columns.
     * @param {Number} width
     */
    setWidth(width) {
        this.width = width;
    }

    getPrefWidth() {
        return this.prefWidth > 0 ? this.prefWidth : this.width;
    }

    setPrefWidth(prefWidth) {
        this.prefWidth = prefWidth;
    }

    getUnits() {
        return this.units;
    }

    setUnits(units) {
        this.units = units;
    }

    getType() {
        return this.type;
    }

    setType(type) {
        this.type = type;
    }

    isSortable() {
        return this.isSortable;
    }

    setSortable(sortable) {
        this.isSortable = sortable;
    }

    isHidden() {
        return this.isHidden;
    }

    setHidden(hidden) {
        this.isHidden = hidden;
        this.isVisible = hidden ? false : this.isVisible;
    }

    isVisible() {
        return this.isVisible;
    }

    setVisible(visible) {
        this.isVisible = visible;
    }

    // @param {Array} enumVals - an array of strings
    setEnums(enumVals) {
        this.enumVals = enumVals;
    }

    getEnums() {
        return this.enumVals;
    }

    // @return {Array} an array of strings
    getSortByCols() {
        return this.sortByCols;
    }

    // @param secondarySortCols - an array of strings
    setSortByCols(secondarySortCols) {
        this.sortByCols = secondarySortCols;
    }

    isRequiresQuotes() {
        return this.requiresQuotes;
    }

    setRequiresQuotes(requiresQuotes) {
        this.requiresQuotes = requiresQuotes;
    }
}
